sdks/python/apache_beam/runners/worker/log_handler.py

```diff
@@ -156,9 +174,11 @@ def _write_log_entries(self):
         done = True
         log_entries.pop()
       if log_entries:
-        yield beam_fn_api_pb2.LogEntry.List(log_entries=log_entries)
+        yield beam_fn_api_pb2.LogEntry.List(
+            log_entries=cast(List[beam_fn_api_pb2.LogEntry], log_entries))
```

Comment:
we have to cast here because `log_entries` started out life as a `List[Union[..., Sentinel]]`, and just above this line we checked for the sentinel and popped it out, but mypy can't track that.